src/backend/backend.c

```diff
@@ -204,15 +204,17 @@ void paint_all_new(session_t *ps, struct managed_win *t, bool ignore_damage) {
 			if (w->state == WSTATE_MAPPING) {
 				// Gradually increase the blur intensity during
 				// fading in.
-				blur_opacity = w->opacity / w->opacity_target;
+				blur_opacity = w->opacity * w->opacity_target;
 			} else if (w->state == WSTATE_UNMAPPING ||
 			           w->state == WSTATE_DESTROYING) {
 				// Gradually decrease the blur intensity during
 				// fading out.
 				blur_opacity =
-				    w->opacity / win_calc_opacity_target(ps, w, true);
```

Comment:
Same as above. Here `w->opacity` goes from `win_calc_opacity_target(ps, w, true)` to `0`.